src/cpp/middleware/fastdds/FastDDSMiddleware.cpp

```diff
@@ -710,7 +724,20 @@ bool FastDDSMiddleware::read_reply(
    auto it = requesters_.find(requester_id);
    if (requesters_.end() != it)
    {
-       rv = it->second->read(sequence_number, data, timeout);
+       fastdds::dds::SampleInfo sample_info;
+       rv = it->second->read(sequence_number, data, timeout, sample_info);
+
+       if (intraprocess_enabled_)
+        {
+            for (auto rp = repliers_.begin(); rp != repliers_.end(); rp++)
+            {
+                if (rp->second->guid_datawriter() == sample_info.sample_identity.writer_guid())
+                {
+                    rv = false;
+                    break;
+                }
+            }
+        }
```

Comment:
This should not be necessary since the proxy client request are blocked when `intraprocess_enabled_` enabled